docs/generating_tasks.md

```diff
@@ -200,6 +200,24 @@ burn_in_tag_compile_distro: rhel80-large
 burn_in_tag_compile_task_group_name: compile_and_archive_dist_test_TG
 ```
 
+`burn_in_tasks_gen` task is used to generate several copies of the task. The example of task
+configuration:
+
+```yaml
+- <<: *gen_burn_in_task_template
+  name: burn_in_tasks_gen
+  tags: []
+  commands:
+  - func: "generate resmoke tasks"
+```
+
+`burn_in_task_name` buildvariant expansion is used to configure which task to burn-in. The
+example of `burn_in_task_name` buildvariant expansion:
+
+```yaml
+burn_in_task_name: replica_sets_jscore_passthrough_gen
```

Comment:
`_gen` tasks are a bit scary to me for us to support as part of `burn_in_tasks`. They're usually very large and are meant to be split up by the generator. I don't think we should split up these tasks as part of `burn_in_tasks` as that will have us wind up creating 50 different tasks in Evergreen.

Can we not allow any tasks that end in `_gen`? It seems likely that they would time out anyway for the most part as a lot of tasks are split up to avoid running forever in patches. Or maybe we should add this point as a warning in our docs to users. Do you have any thoughts @MikhailShchatko ?

Reply:
I think we can add a warning message and something in docs, that some tasks may time out, since task splitting is not supported in burn-in tasks. @zamj what do you think?

Reply:
That sounds good to me

Reply:
Added information in docs.

Now I think it's not very useful to print a message by task generator since it's not very likely to me that somebody will look at `version_burn_in_gen` logs when their tasks will be timing out.

I think it's better to add a comment in mongo's `evergreen.yml` above `burn_in_task_name` expansion with the message.

@zamj let me know what you think?

Reply:
I think that sounds good to me, I agree it's likely more visible there.